### DISCUSSION TITLE
How to re-flash bricked ZuluSCSI using DFU-UTIL

### DISCUSSION THREAD

**Discussion by scalarmapedia** [ZuluSCSI/ZuluSCSI-firmware#51] [Q&A] [answered]
Hey, I have bricked my ZuluSCSI 1.1 by flashing it with incorrect .bin (I know...) anway, the docs state you can get the device on its feet again by flashing it using DFU-UTIL, and thats about it... no examples, what file should be used, etc. Is there ANY kind of documentation about this?

**Comment by aperezbios** [comment, marked as answer]
@scalarmapedia Thanks for reaching out. I can provide you with the information that you need, but it's platform-specific. Which platform(s)/Operating Systems do you have access to? Because of the need to install a driver for DFU under Windows, it's generally less of a hassle if you're able to run dfu-util on a Linux or Mac based system.

**Reply by scalarmapedia** [reply]
I have a Windows and Mac system. So whichever is easier, will do me. I installed a dfu-util on windows, but it seems much older than the one installed on MacOS using Brew

**Reply by aperezbios** [reply]
@scalarmapedia All you need to do is flip the BOOT DIP switch on, which will put the board in DFU mode when it powers on. You can switch it off within a second, as the state of the DIP only matters at initial power on. it MUST be switched to off for the firmware to load after you've re-flashed via DFU.

You should just need to run 'dfu-util -a 0 --dfuse-address 0x08000000 -D /path/to/right/firmware.bin'

If for whatever reason that fails, you can try under Windows. You'll need to download these two assets:

The GigaDevice programming utility requires you to install a USB driver in Windows first, which you can get from here:
https://gd32mcu.com/download/down/document_id/244/path_type/1
…and then their actual DFU tool is here:
https://gd32mcu.com/download/down/document_id/176/path_type/1

**Reply by scalarmapedia** [reply]
I ran the above command on dfu-util on MacOS and it worked! So, thanks for that. So, now my Zulu is back on its feet, and with 1.0.7 installed. 

Sadly, its not fixed it, not booting with my Amiga 2000's TrumpCard Professional. Which does a Software error during boot, and then re-boots, reapeat. Are there any forums or whatnot where I can get help with this.
The Zulu DOES work when attached to this SCSI adapter, its just this final step, to get it auto-booting that is frustrating me so :/

**Comment by aperezbios** [comment]
You're already in the right place. This is a ZuluSCSI-specific discussion area. Try switching the DEBUG DIP switch, and power-cycling your ZuluSCSI/Amiga. Then attempt to (auto)boot, and attach the zululog.txt file from the SD card to this discussion thread, as an attachment.

**Reply by scalarmapedia** [reply]
Sure, here is the log. What you are seeing there is several power cycles of the Amiga as it attempts to boot, Software errors and the re-boots. Repeat... To be clear, the HD0: drive on the SD has a fully installed Workbench hard disk, setup using IVS utils, specifically for this the TrumpCard Pro. Originally this machine had a real SCSI drive in it, with the same software on it, and it was booting.
[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/9501739/zululog.txt)

**Reply by PetteriAimonen** [reply]
Hmm, nothing obviously wrong in the log. The Amiga issues a read to sector 55, which completes ok and then it just restarts.

Could there possibly be some problem with the partition table stored inside the image file?

**Reply by scalarmapedia** [reply]
Well, this IS different to earlier behaviour, previously, with the TrumpCards Autoboon ON, the device would just hang, so at least now we can see that the Amiga is asking data from a certain sector. So, I guess there could be a few scenarios at play here...

1. The Amiga wants to boot from Sector 55, which in the emulated drive does not contain the right data, hence the Software Error and reboot
2. The drive tells the Amiga that the boot should happen from Sector 55, and the incorrect data is there. Boom,crash.
3. The Workbench on the drive is not quite correct, version or whatnot, it IS loading and then causing crash.

I DO have a real SCSI drive, which was on this device and did boot. I was in the action of backing it up, when the machine became unstable, I resolved this but have not booted from that drive since. I guess I COULD try getting it going, back it up onto a Suitable ZuluSCSI device, and then try and boot from it.

Not sure which path to follow. Any advice?

**Reply by scalarmapedia** [reply]
So, I tried following path 3. I backed up the physical drive (Quantum ProDrive LPS 100mb) that DOES boot. Then I created a new HD0 and set it up using the Util that came with the SCSI adapter. Then I formatted it in Workbench, and copied over all the files from the original boot disk. This HDO was created AS DH0 in an emulator (Amiga Forever), and is named HD0.hda, so all that syncs up.
and....
It crashes just like before during boot :/

**Comment by aperezbios** [comment]
@scalarmapedia another debug log couldn't hurt.

**Comment by PetteriAimonen** [comment]
Googling gives some thread with similar problems with TrumpCard and SCSI2SD: https://eab.abime.net/showthread.php?p=1549792
On the other hand this video seems to successfully boot the same combination: https://www.youtube.com/watch?v=xapfFd_v5vg

Reducing speed settings may change something also, try e.g. this `zuluscsi.ini`:

```
[SCSI]
PhyMode = 1
SelectionDelay = 1
PrefetchBytes = 0
EnableSCSI2 = 0
```
